### PR TITLE
Fixes #3117 and broken links

### LINKED ISSUE #3117: Knative Serving(docs/serving/_index.md)
## Expected Behavior
Reference to generic networking layer

## Actual Behavior
Reference to Istio. Depending on the installation choice, this may or may not be accurate.

## Steps to Reproduce the Problem
Visit [here](https://knative.dev/docs/serving/)

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #3117
Fixes broken / incorrect links

### CODE REVIEW

**Comment by abrennan89** [comment]
/approve
/hold

Hold til someone from eng (@evankanderson ?) gives it the OK, just in case

**Review by abrennan89 — COMMENTED** [review]
/unhold

File: docs/serving/_index.md [hunk 1]
```diff
@@ -5,39 +5,37 @@ type: "docs"
...
-of serverless applications and functions. Serving is easy to get started with
-and scales to support advanced scenarios.
+Knative Serving builds on Kubernetes to support deploying and serving serverless applications and functions. Knative Serving is easy to get started with and scales to support advanced scenarios.
 
-The Knative Serving project provides middleware primitives that enable:
+The Knative Serving project provides middleware components that enable:
 
-- Rapid deployment of serverless containers
-- [Automatic scaling up and down to zero](./autoscaling/)
-- Routing and network programming for Istio components
```
Comment (evankanderson) [inline docs/serving/_index.md:16]: Haha, wow, thanks!

File: docs/serving/spec/knative-api-specification-1.0.md [hunk 2]
```diff
@@ -1,3 +0,0 @@
-# Knative Serving API Specification
-
-This file has been moved to the [Knative Specs Repository](https://github.com/knative/specs/blob/main/specs/serving/knative-api-specification-1.0.md)
```
Comment (evankanderson) [inline docs/serving/spec/knative-api-specification-1.0.md:3]: I think we left this file as a pointer for external links. I'm not seeing the redirect right now, so we might want to leave this in place?

https://github.com/knative/website/blob/main/static/_redirects#L3

**Review by evankanderson — COMMENTED** [review]
Not sure if we want to add the stub file back in or figure out the redirect.

/lgtm
/approve

**Pull request merged**